### PR TITLE
typescript: activityDetection unit test

### PR DESCRIPTION
**Pull request by vrbsm**
## Description
activityDetection unit test

## QA Checklist
- [x] Unit-Tests
- [ ] Automation-Tests

Attach unit & spectron tests results in PDF format against the above task lists for this branch

### CODE REVIEW

File: spec/activityDetection.spec.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+        const spy = jest.spyOn(activityDetection, 'setWindowAndThreshold');
+
+        activityDetection.setWindowAndThreshold({}, 1000);
+
+        expect(spy).toBeCalledWith({}, 1000);
+    });
+
+    it('should start activity monitor when `setWindowAndThreshold` is called', () => {
+        const { activityDetection } = require('../src/app/activity-detection');
```
Comment (VishwasShashidhar) [inline spec/activityDetection.spec.ts]: any reason we are requiring here? Can't we use import at the beginning of the file?

File: spec/activityDetection.spec.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        jest.useFakeTimers();
+    });
+
+    afterAll((done) => {
+        jasmine.DEFAULT_TIMEOUT_INTERVAL = originalTimeout;
+        done();
+    });
+
+    it('should call `setWindowAndThreshold` correctly', () => {
+        const { activityDetection } = require('../src/app/activity-detection');
```
Comment (VishwasShashidhar) [inline spec/activityDetection.spec.ts]: any reason we are requiring here? Can't we use import at the beginning of the file?

File: spec/activityDetection.spec.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        const spy = jest.spyOn(activityDetection, 'startActivityMonitor')
+        .mockImplementation(() => jest.fn());
+
+        activityDetection.setWindowAndThreshold({}, 1000);
+
+        expect(spy).toBeCalled();
+    });
+
+    it('should call `activity` when `startActivityMonitor` is called', () => {
+        const { activityDetection } = require('../src/app/activity-detection');
```
Comment (VishwasShashidhar) [inline spec/activityDetection.spec.ts]: any reason we are requiring here? Can't we use import at the beginning of the file?

File: spec/activityDetection.spec.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        const electron = require('./__mocks__/electron');
```
Comment (VishwasShashidhar) [inline spec/activityDetection.spec.ts]: any reason we are requiring here? Can't we use import at the beginning of the file?

File: spec/activityDetection.spec.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+        activityDetection.startActivityMonitor();
+
+        jest.runOnlyPendingTimers();
+
+        expect(spy).toBeCalledWith(15);
+    });
+
+    it('should call `sendActivity` when period was greater than idleTime', () => {
+        const { activityDetection } = require('../src/app/activity-detection');
```
Comment (VishwasShashidhar) [inline spec/activityDetection.spec.ts]: any reason we are requiring here? Can't we use import at the beginning of the file?

File: spec/activityDetection.spec.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        // period is this.idleThreshold = 60 * 60 * 1000;
+        const spy = jest.spyOn(activityDetection, 'sendActivity');
+
+        activityDetection.activity(50);
```
Comment (VishwasShashidhar) [inline spec/activityDetection.spec.ts]: Please move the value `50` to a variable.

File: spec/activityDetection.spec.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+        expect(spy).toBeCalledWith(50 * 1000);
```
Comment (VishwasShashidhar) [inline spec/activityDetection.spec.ts]: Please move the values `50` & `1000` to a variable.

File: spec/activityDetection.spec.ts (hunk 1 above)
Comment (vrbsm) [reply spec/activityDetection.spec.ts]: Basically I did it to reset modules: 
https://github.com/facebook/jest/issues/3236#issuecomment-290849912
https://jestjs.io/docs/en/jest-object#jestresetmodules
but I changed according your comment

**Pull request merged**